src/main.cpp

```diff
@@ -1102,23 +1102,32 @@ bool ContextualCheckTransaction(
                     error("ContextualCheckTransaction(): consensus branch id not set"),
                     REJECT_INVALID, "bad-tx-missing-consensus-branch-id");
             }
+
+            // v5 transactions must have empty joinSplits
```

Comment:
I think it's fine to put this consistency check here. If NU5 is not active, then v5 transactions will be unilaterally rejected, and if NU5 is active, then we reach this check.